src/OpenTelemetry.Api/Context/CorrelationContext.cs

```diff
@@ -0,0 +1,76 @@
+﻿// <copyright file="CorrelationContext.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenTelemetry.Context
+{
+    /// <summary>
+    /// Correlation context.
+    /// </summary>
+    public readonly struct CorrelationContext : IEquatable<CorrelationContext>
+    {
+        private static readonly List<CorrelationContextEntry> EmptyList = new List<CorrelationContextEntry>();
```

Comment:
>private static readonly List<CorrelationContextEntry> EmptyList = new List<CorrelationContextEntry>(); [](start = 8, length = 102)

nit: I am seeing this is used only in CorrelationContext Empty { get; } so I am not sure if it worth to declare it. 

you can just have.

public static CorrelationContext Empty { get; } = new CorrelationContext(new  List<CorrelationContextEntry>(););

it is ok to keep it if you think we are going to use it for other purpose.